### PR TITLE
PROD-610: add associated address if doesn't exists, optimize send bonk an…

### PR DESCRIPTION
**Pull request by yash431garg**
- Description
Previously it wasn't possible to add bonk to a new wallet. 
- What are the steps to test that this code is working?
Create a new wallet and try to claim bonk. It should be succesfull.
- Screen shots or recordings for UI changes
N/A

### CODE REVIEW

**Comment by yash431garg** [comment]
@marvinmarnold I have rebased this branch with main and squashed as well. If you think rebase is correct we can merge this work and close #798

File: app/utils/claim.ts [hunk 1]
```diff
@@ -71,24 +67,40 @@ export const sendBonk = async (
 
   const bonkMint = new PublicKey(process.env.NEXT_PUBLIC_BONK_ADDRESS!);
 
-  const fromTokenAccount = await getAssociatedTokenAddress(
+  const treasuryATA = await getAssociatedTokenAddress(
```
Comment (neel-ds) [inline app/utils/claim.ts:70]: Can you use more descriptive name like treasuryTokenAccount?

File: app/utils/claim.ts [hunk 2]
```diff
@@ -71,24 +67,40 @@ export const sendBonk = async (
...
     bonkMint,
     fromWallet.publicKey,
   );
-  const toTokenAccount = await getAssociatedTokenAddress(bonkMint, toWallet);
 
-  const instruction = createTransferInstruction(
-    fromTokenAccount,
-    toTokenAccount,
+  const receiverATA = await getAssociatedTokenAddress(bonkMint, toWallet);
```
Comment (neel-ds) [inline app/utils/claim.ts:75]: same goes here with name

File: app/utils/claim.ts [hunk 3]
```diff
@@ -71,24 +67,40 @@ export const sendBonk = async (
...
+    );
+    instructions.push(ataInstruction);
+  }
+
+  const transferInstruction = createTransferInstruction(
+    treasuryATA,
+    receiverATA,
     fromWallet.publicKey,
     amount,
+    [],
```
Comment (neel-ds) [inline app/utils/claim.ts:97]: Can you add a comment explaining the empty array argument?

File: app/utils/claim.ts [hunk 4]
```diff
@@ -113,16 +125,16 @@ export const sendBonk = async (
   instructions.unshift(computeBudgetIx);
 
   //refered from past transaction
-  const computeUnitFix = 4794;
-  // Buffer to make sure the transaction doesn't fail because of less compute units
-  const computeUnitsIx = ComputeBudgetProgram.setComputeUnitLimit({
-    units: Math.round(computeUnitFix * 1.1),
-  });
+  // const computeUnitFix = 4794;
```
Comment (neel-ds) [inline app/utils/claim.ts:128]: is it fine to remove CU optimization here if it is not required?

File: app/utils/claim.ts (hunk 1 above)
Comment (yash431garg) [reply app/utils/claim.ts:70]: It is not a token account it is AssociatedTokenAddress which means it is the mint address for that account. If you want I can make it treasuryAssociatedTokenAddress.

File: app/utils/claim.ts (hunk 3 above)
Comment (yash431garg) [reply app/utils/claim.ts:97]: Sure,

File: app/utils/claim.ts (hunk 4 above)
Comment (yash431garg) [reply app/utils/claim.ts:128]: Yea, I have about will ask in the call regarding this.

**Review by marvinmarnold — APPROVED** [review]

**Review by neel-ds — APPROVED** [review]

**Pull request merged**